ibis/expr/tests/test_interactive.py

```diff
@@ -30,6 +30,12 @@ def test_interactive_execute_on_repr(self):
 
         assert len(self.con.executed_queries) > 0
 
+    def test_png_repr_returns_correct_type(self):
```

Comment:
can you add a test when interactive is False a well

Reply:
Good point. I tried that in the past, and it's not trivial. There are different cases that affect the output of `_repr_png_`. Like, graphviz not being installed, anything that makes importing `ibis.expr.visualize` fail, or call the `to_graph` method. Also if the `graphviz_repr` option is set to `False`.

I added the test, to see what fails, but I think it'll be tricky to find a way to add that test in a deterministic and reliable way. We may end up introducing a test that doesn't provide much value, but it's flaky when new builds are added.

Reply:
right its just to verify that it is NOT none

Reply:
Yep, but the problem is that even with `interactive` set to `False`, `_repr_png_` can still be `None` for different reasons:
- graphviz not installed
- option `graphviz_repr` set to `False`
- an exception happens when importing the visualization module

I added a test for the case when all conditions are met for the result to be `not None`. It's a bit tricky, but I think it's the best we can do.